docs/zenith/developer/go/457202-test-build-publish.md

```diff
@@ -0,0 +1,314 @@
+---
+slug: /zenith/developer/go/457202/test-build-publish
+displayed_sidebar: "zenith"
+toc_max_heading_level: 2
+authors: ["Erik Sipsma"]
+date: "2023-11-07"
+---
+
+# Test, Build and Publish an Application using a Module
+
+## Introduction
+
+This guide walks you through the process of creating a Dagger module from scratch to test, build and publish a Node.js application image. You will learn how to:
+
+- Initialize a new Dagger module and add dependencies to it
+- Write custom functions using the Go SDK to test, build and package an application
+- Use containers as function input arguments and return values
+- Debug module code using an interactive shell
+- Test module code by exposing services or containers returned by a function on the host
+
+## Requirements
+
+This guide assumes that:
+
+- You have a good understanding of the Go programming language and an IDE with Go language support.
+- You know the basics of programming Dagger modules. If not, refer to the [quickstart](./525021-quickstart.md).
+- You have the Dagger CLI installed in your development environment. If not, [install the Dagger CLI](../../../current/cli/979595-reference.md).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+- You have a Node.js Web application. If not, follow the steps in Appendix A to [create an example Node.js application](#appendix-a-create-an-example-application).
```

Comment:
I'd like to make this optional by just having a default in the module that uses a node project from git if the user doesn't specify one. But not worth blocking on; I'll update this once I have time to do that.

Reply:
OK thanks!